mbus/mbus-tcp.c

```diff
@@ -52,34 +49,34 @@ mbus_tcp_connect(mbus_handle *handle)
         return -1;
 
     host = tcp_data->host;
-    port = tcp_data->port;
+    sprintf(port, "%d", tcp_data->port);
```

Comment:
The usage of snprintf is safer here. Also please check the type of port with your format specifier.